fix for last open items at https://github.com/OWASP/owasp-masvs/issues/203 (#1479)
OWASP/owasp-mstg: Document/0x05i-Testing-Code-Quality-and-Build-Settings.md

Several best practices for [configuring the app for release](https://developer.android.com/tools/publishing/preparing.html#publishing-configure "Best Practices for configuring an Android App for Release") are available in the official Android developer documentation.

> Last, but not least: make sure that the application is never deployed with your internal testing certificates.

Reviewer: ```suggestion
Last but not least: make sure that the application is never deployed with your internal testing certificates.
```

- Libraries that are not (or should not) be packed within the actual production application, such as `Mockito` used for testing and libraries like `JavaAssist` used to compile certain other libraries.
- Libraries that are packed within the actual production application, such as `Okhttp3`.

These libraries can lead to unwanted side-effects:

- A library can contain a vulnerability, which will make the application vulnerable. A good example are the versions of `OKHTTP` prior to 2.7.5 in which TLS chain pollution was possible to bypass SSL pinning.

> - A library can no longer be maintained or hardly be used, which is why no vulnerabilities are reported and/or fixed. This can lead to having bad and/or vulnerable code in your application through the library.

Reviewer 2: Slightly re-phrased. What do you think? 

```suggestion
- A library is no longer maintained or is not widely used, which is why no vulnerabilities are reported and/or fixed. This can lead to having bad and/or vulnerable code in your application because of that library.
```

> These libraries can have the following two classes of unwanted side-effects:

Reviewer: ```suggestion
These libraries can lead to unwanted side-effects:
```

When the sources are not available, one can decompile the app and check the JAR files. When Dexguard or Proguard are applied properly, then version information about the library is often obfuscated and therefore gone. Otherwise you can still find the information very often in the comments of the Java files of given libraries. Tools such as MobSF can help in analyzing the possible libraries packed with the application. If you can retrieve the version of the library, either via comments, or via specific methods used in certain versions, you can look them up for CVEs by hand.

> If the application is a high risk application, you will end up vetting the library manually. In that case, there are specific requirements for native code, which you can find in the chapter ["Testing Code Quality"](0x04h-Testing-Code-Quality.md). Next to that, it is good to vet whether all best practices for software engineering are applied.

Reviewer 2: high-risk + Normally we put the quotes outside the markdown link.

```suggestion
If the application is a high-risk application, you will end up vetting the library manually. In that case, there are specific requirements for native code, which you can find in the chapter "[Testing Code Quality](0x04h-Testing-Code-Quality.md)". Next to that, it is good to vet whether all best practices for software engineering are applied.
```

Author: Good point, fixed.

> If the application is a high risk application, you will end up vetting the library manually. In that case there are a wild amount of review requirements, which are similar to the requirements established by the MASVS for the applicaiton as a whole. Next to that, it is good to vet whether all best practices for software engineering are applied.

Reviewer: ```suggestion
If the application is a high risk application, you will end up vetting the library manually. In that case there are specific requirements for native code, which are similar to the requirements established by the MASVS for the application as a whole. Next to that, it is good to vet whether all best practices for software engineering are applied.
```
+ Should we reference some sort of low level programming checklist?

Reviewer 2: Comma missing? -> In that case,

Author: Made a link to chapter 4 on code quality

> If the application is a high risk application, you will end up vetting the library manually. In that case, there are a specific requirements for native code, which you can find in the chapter ["Testing Code Quality"](0x04h-Testing-Code-Quality.md). Next to that, it is good to vet whether all best practices for software engineering are applied.

Reviewer 3: Remove „a“ before specific

> ##### Detecting the licenses used by the libraries of the application

Reviewer 2: ```suggestion
##### Detecting the Licenses Used by the Libraries of the Application
```
